spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/encoding/EnableFeignAcceptGzipEncoding.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2013-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.netflix.feign.encoding;
+
+import org.springframework.context.annotation.Import;
+
+import java.lang.annotation.*;
+
+/**
+ * Enables receiving GZIP/Deflate compressed responses by sending in response header {@code Accept-Encoding}
+ * {@code gzip} {@code deflate} as supported encoding types. Though, this not yet implies that the response
+ * will be compressed and requires the server to be configured for compressing the output responses.
+ *
+ * @author Jakub Narloch
+ * @see FeignAcceptGzipEncodingInterceptor
+ * @see FeignAcceptGzipEncodingConfiguration
+ */
+@Documented
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Import(FeignAcceptGzipEncodingConfiguration.class)
+public @interface EnableFeignAcceptGzipEncoding {
```

Comment:
Can we go with an auto-configuration enabled via properties, rather than an @Enable? Something similar to what spring-boot [does for tomcat](http://docs.spring.io/spring-boot/docs/current/reference/html/howto-embedded-servlet-containers.html#how-to-enable-http-response-compression).